packages/flutter_tools/lib/src/ios/devices.dart

```diff
@@ -300,25 +300,25 @@ class IOSDevice extends Device {
       return new LaunchResult.failed();
     }
 
-    return new LaunchResult.succeeded(observatoryPort: localObsPort, diagnosticPort: localDiagPort);
+    return new LaunchResult.succeeded(observatoryUri: localObsUri, diagnosticUri: localDiagUri);
   }
 
-  Future<int> _acquireAndForwardPort(
+  Future<Uri> _acquireAndForwardPort(
```

Comment:
this function is named '_acquireAndForwardPort' and it should continue to return a port. The caller can then construct a Uri from it (if necessary).